src/libs/antares/study/parallel-years.h

```diff
@@ -0,0 +1,165 @@
+
+    // Associe le numero d'une année jouée à l'indice de l'espace
+    std::map<unsigned int, unsigned int> performedYearToSpace;
+
+    // L'inverse : pour une année jouée, associe l'indice de l'espace au numero de l'année
+    std::map<unsigned int, unsigned int> spaceToPerformedYear;
+
+    // Pour chaque année, est-elle la première à devoir être jouée dans son lot d'années ?
+    std::map<unsigned int, bool> isFirstPerformedYearOfASet;
+
+    // Pour chaque année du lot, est-elle jouée ou non ?
+    std::map<unsigned int, bool> isYearPerformed;
+
+    // Nbre d'années en parallele vraiment jouées pour ce lot
+    unsigned int nbPerformedYears;
+
+    // Nbre d'années en parallele jouées ou non pour ce lot
+    unsigned int nbYears;
+
+    // Regenere-t-on des times series avant de jouer les annees du lot courant
+    bool regenerateTS;
+
+    // Annee a passer a la fonction "regenerateTimeSeries<false>(y)" (si regenerateTS is "true")
+    unsigned int yearForTSgeneration;
+};
+
+/*!
+** \brief Temporary Area List Holder
+**
+** \warning TODO: REMOVE THIS CLASS ONCE THE TSGEN HAS BEEN DECOUPLED FROM SOLVER
+**
+** To compute the sets of parallel years while using the Time Series Generator
+** we need to know in advance if the area's thermal clusters will be refresehd during generation
+** Although, since the max number of parallel years is needed to create the areas, this info
+** is not available yet (and tht's why, originally, this calculation was done twice)
+**
+** This temporary holder will load the area list and will check for each one if
+** their clusters will require refreshing during generation
+*/
+class TempAreaListHolder
+{
+public:
+    TempAreaListHolder() = default;
+
+    bool checkThermalTSGeneration(YString folder_);
+
+private:
+    void loadAreaList();
+    YString inputFolder;
+    std::vector<Yuni::String> areaNames;
+
+};
+
+class SetsOfParallelYearCalculator
+{
+public:
+    
+    SetsOfParallelYearCalculator(bool forceParallel, bool enableParallel, uint forcedNbOfParallelYears,
+                                 uint number_of_cores, bool thermalTSRefresh, Parameters &params)
+    : forceParallel_{forceParallel},
+    enableParallel_ {enableParallel},
+    forcedNbOfParallelYears_{forcedNbOfParallelYears},
+    number_of_cores_{number_of_cores},
+    thermalTSRefresh_{thermalTSRefresh},
+    p{params}{
+        this->build();
+    }
+
+        
+    bool allSetsParallelYearsHaveSameSize();
+
+    [[nodiscard]] uint getForcedNbOfParallelYears() const
+    {
+        return forcedNbOfParallelYears_;
+    } 
+
+    [[nodiscard]] uint getMinNbParallelYearsForGUI() const
+    {
+        return computeMinNbParallelYears();
+    }
+
+    [[nodiscard]] std::vector<setOfParallelYears> getSetsOfParallelYears() const
+    {
+        return setsOfParallelYears;
+    }
+
+    [[nodiscard]] uint getNbYearsReallyPerformed() const
+    {
+        return nbYearsReallyPerformed;
+    }
+
+private:
+    
+    void build();
```

Comment:
About the **build()** method : 
- Make the **build()** public
- We should not call it from the constructor
- but call it from outside, after an object of type **SetsOfParallelYearCalculator** is instantiated